misc/python/materialize/parallel_workload/execute.py

```diff
@@ -0,0 +1,89 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import threading
+
+import pg8000
+
+
+def execute(cur: pg8000.Cursor, query: str) -> bool:
+    query += ";"
+    name = threading.current_thread().getName()
+    try:
+        try:
+            cur.execute(query)
+        except Exception as e:
```

Comment:
I think all errors should just result in a `ROLLBACK` and moving on to the next generated statement. There is no need to reset the isolation level and retry the statement, as I think this has the potential to confuse developers going forward -- like if in the future the transaction isolation level is made randomized or configurable, this method is going to randomly reset it to some other value.